### PR TITLE
CS Implement

### PR DESCRIPTION
**Pull request by TomChermin**
Added functionality for CS
I think it works now and no longer has negative bounty possibilities

### CODE REVIEW

File: Strings.ts [hunk 1]
```diff
@@ -177,4 +179,19 @@ export class Strings {
     static placedBounty(bountyPlacer: string, bounty: number, bountyTarget: string): string {
         return `📃 @${bountyPlacer} put a bounty worth ${bounty} points on @${bountyTarget} !`;
     }
+
+    static currentlyCSuserName: string | null = null): string {
```
Comment (MaikoKingma) [inline Strings.ts:183]: missing an ( and no parameter name

File: Strings.ts [hunk 2]
```diff
@@ -132,6 +133,7 @@ export class Strings {
         + `/${Commands.placebounty} - Place a bounty on a player\n`
         + `/${Commands.kill} - Attempt to kill a player to claim their bounty\n`
         + `/${Commands.togglelifetags} - Toggles if you are tagged whenever you are done working or completed your sentence`;
+        + `/${Commands.communityService} - Perform community service to reduce your bounty\n`;
```
Comment (MaikoKingma) [inline Strings.ts:136]: you forgot to add this command to the Commands.ts file

File: plugin.ts [hunk 3]
```diff
@@ -58,8 +58,9 @@ export class Plugin extends AbstractPlugin {
     const bountiesCommand = new BotCommand([Commands.bounties], "", this.bounties.bind(this), false);
     const placeBountyCommand = new BotCommand([Commands.placebounty], "", this.placeBounty.bind(this), false);
     const killPlayerCommand = new BotCommand([Commands.kill], "", this.kill.bind(this), false);
+    const communityServiceCommand = new BotCommand([Commands.cs1, Commands.cs2], "", this.communityService.bind(this), false);
```
Comment (MaikoKingma) [inline plugin.ts:61]: you forgot to add these commands to the Commands.ts file

File: plugin.ts [hunk 4]
```diff
@@ -401,4 +402,50 @@ export class Plugin extends AbstractPlugin {
...
+      }
+      if (minutes > 60) {
+        return `Maybe start actually improving the community rather than spending time on afk 😤`;
+      }
+    } else {
+      minutes = Random.number(2, 10);
+    }
+    const multiplier: number = chat.getSetting(Strings.CS_MULTIPLIER_SETTING);
+
+    lifeUser.startCommunityService(minutes, () => {
```
Comment (MaikoKingma) [inline plugin.ts:428]: lifeChatData, targetUser and bounty are not defined in this function

File: plugin.ts [hunk 5]
```diff
@@ -401,4 +403,53 @@ export class Plugin extends AbstractPlugin {
...
+     const parameters = match.split(' ');
```
Comment (MaikoKingma) [inline plugin.ts:430]: `match` is undefined here when a user does `/cs` without tagging another user (eg. applying the cs on himself)

Comment (MaikoKingma) [reply plugin.ts:430]: sidenote, I'm guessing the `match` variable is expected to contains some kind of user tag? I have never used `match` and don't know whats its for.

Comment (TomChermin) [reply plugin.ts:430]: I've also never used match and don't know what it's for :P

**Comment by TomChermin** [comment]
Move bounty check to the start of cs instead of the resolve

**Comment by TomChermin** [comment]
Add a status for cs

**Review by Agadar — COMMENTED** [review]
Let me know when this is fully tested.

File: plugin.ts [hunk 6]
```diff
@@ -401,4 +412,55 @@ export class Plugin extends AbstractPlugin {
...
+      if (minutes > 60) {
+        return `Maybe start actually improving the community rather than spending time on afk 😤`;
+      }
+    } else {
+      minutes = Random.number(2, 10);
+    }
+    const multiplier: number = chat.getSetting(Strings.CS_MULTIPLIER_SETTING);
+
+    lifeUser.startCommunityService(minutes, () => { //The actual community service
+     const lifeChatData = this.helper.getOrCreateLifeChatsData(chat.id);
```
Comment (MaikoKingma) [inline plugin.ts:445]: this line is needed before line 418 otherwise lifeChatData is undefined there

File: plugin.ts [hunk 7]
```diff
@@ -401,4 +412,54 @@ export class Plugin extends AbstractPlugin {
...
+      }
+      const multiplier: number = chat.getSetting(Strings.CS_MULTIPLIER_SETTING);
+
+      lifeUser.startCommunityService(minutes, () => { //The actual community service
+    
+      let scoreToGain = 0;
+
+      //Reduces the bounty by x amount
+      scoreToGain += lifeUser.occupation!.waitingTime * 20 * multiplier;
+      chatBounty.bounty += scoreToGain * (-1);
```
Comment (MaikoKingma) [inline plugin.ts:453]: `scoreToGain = scoreToGain > chatBounty ? chatBounty : scoreToGain`

**Comment by Agadar** [comment]
What's the status on this @TomChermin @MaikoKingma ?

**Pull request closed without merging**